src/pass.h

```diff
@@ -199,6 +199,11 @@ struct PassRunner {
   // afterwards.
   void addDefaultGlobalOptimizationPostPasses();
 
+  // Adds optimizations that should only be run immediately prior to module
+  // writing. After these passes the module may be in Poppy IR. This is not
```

Comment:
Poppy IR reference seems to be jumping the gun?

Reply:
Oh right, sorry. I refactored this out of my local branch because it is something we will have to do no matter what solution we end up with. Merging this now (with the comment fixed) is a nice incremental change that makes follow-on diffs simpler.

Reply:
I'm unsure why we need this no matter what solution we end up with?

Another option is to run final passes before binary writing from the binary writing code. That would avoid the risk of forgetting them somewhere. But it feels a little early for that as well, to me - my feeling is it's best to decide on the direction first, then things will wall out of that.

Reply:
You're right, that would be another option. I'll mark this PR as draft for now since we want to wait for more information before merging.